hw/BUILD

```diff
@@ -1,3 +1,87 @@
 # Copyright lowRISC contributors.
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
+
+load("//rules:fusesoc.bzl", "fusesoc_build")
+
+package(default_visibility = ["//visibility:public"])
+
+fusesoc_build(
+    name = "verilator",
+    srcs = [
+        "//hw:all_files",
+    ],
+    cores = [
+        "//:cores",
+    ],
+    systems = ["lowrisc:dv:chip_verilator_sim"],
+    target = "sim",
+)
+
+filegroup(
```

Comment:
if we add / remove modules in the future, is this something we'll have to manually update?
Because this stuff is also somewhat created by fusesoc right now..so not sure if in the future bazel will replace fusesoc in that regard.  If not though... maybe it makes sense for fusesoc to spit out a list?

Reply:
Yes, the input file list needs to be kept up-to-date.

Since the current `fusesoc_build` action is non-hermetic, I can try to make it not require a `srcs` list... I'm not sure if I can make that work or not.

Reply:
i dont fully understand this syntax... but does `**` include things in subdirectories? I'm not fully understanding why we also need all the individual ip's below, but are missing some such as those in `hw/ip/prim`

Reply:
I think ```**``` includes only files in this directory and in subdirectories without BUILD files. As we add build files we have to point to them explicitly as is done here.

Reply:
ooooo thank you.

Reply:
Yes, `**` includes subdirectories, but, the returned list of labels does not cross "package" boundaries (ie: any subdirectory with a BUILD file.  See [item 4 in the 'limitations' for `glob`](https://docs.bazel.build/versions/main/be/functions.html#glob)).

I need to supply a list of inputs (`srcs`) for this rule, as that is how bazel knows whether or not it needs to rebuild verilator.

Reply:
but i guess maybe that raises another question about why we need the individual IP BUILD files if `**` can include everything without BUILD files? Like i mentioned the list here is already missing some directories under `hw/ip`, so clearly it is already working.

Reply:
We already have some packages to satisfy dependencies for targets in the sw directory; For example, autogenerated header files.

Reply:
The IPs need individual BUILD files because many of them contain an `autogen_hjson_header` rule to create the various `*_regs.h` files needed by software.  This naturally creates a package boundary for any `data` subdirectory with a BUILD rule.

(Currently, I view the `all_files` target is kind-of a hack.  A better long-term approach is to be more precise about required verilog files in a way similar to `cc_library`.  I'm just not ready to approach that in the `hw` hierarchy yet).

Reply:
Appropos of nothing, can we make sure that it's clear this is a hack that we intend to clean up? (Tracking issue etc.)